### PR TITLE
Mt client failover

### PR DESCRIPTION
**Pull request by michalxo**

### CODE REVIEW

**Comment by michalxo** [comment]
Is this waiting on your tests only? Or on what please? Thank you!

**Comment by jdanekrh** [comment]
I approved this and merged as #68 yesterday. I am not sure how you managed to create a new PR that is completely empty (look at "files changed"). I suggest this one be just closed.

**Comment by jdanekrh** [comment]
Ok, now there is new content, but we have conflicts, because this branch contains essentially duplicate commits with the master branch.

@michalxo please rebase on top of master and choose to skip such commits, then force-push (force push is the only way to resolve this, as far as I know).

**Comment by michalxo** [comment]
Did we manage to push this into master or not? :-)
If not, I will have a look at it and push it there.

File: cli-artemis-jms/src/main/java/com/redhat/mqe/acc/AccSenderClient.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
 import com.redhat.mqe.lib.JmsMessageFormatter;
 import com.redhat.mqe.lib.Utils;
 import com.redhat.mqe.lib.message.MessageProvider;
-import org.apache.activemq.artemis.api.core.ActiveMQUnBlockedException;
+import org.apache.activemq.artemis.api.core.*;
```
Comment (jdanekrh) [inline cli-artemis-jms/src/main/java/com/redhat/mqe/acc/AccSenderClient.java]: When do we want to use * import? By default, IntelliJ will put it there if you have at least 3 classes imported from the same package.

I remember that once we had decision always name the imported classes explicitly.

Maybe increase the setting in Intellij, from 3 to 5, 10?

File: cli-artemis-jms/src/main/java/com/redhat/mqe/acc/AccSenderClient.java [hunk 2]
```diff
@@ -99,7 +99,8 @@ public void startClient() {
                     msgProducer.send(message);
                 } catch (JMSException jex) {
                     LOG.error(jex.getCause().toString());
-                    if (jex.getCause() instanceof ActiveMQUnBlockedException) {
+                    if (jex.getCause() instanceof ActiveMQUnBlockedException &&
+                        ((ActiveMQUnBlockedException) jex.getCause()).getType().equals(ActiveMQExceptionType.UNBLOCKED)) {
```
Comment (jdanekrh) [inline cli-artemis-jms/src/main/java/com/redhat/mqe/acc/AccSenderClient.java]: Why do we need `equals(ActiveMQExceptionType.UNBLOCKED)`? The instanceof is not enough?

File: jmslib/src/main/java/com/redhat/mqe/lib/SenderClient.java [hunk 3]
```diff
@@ -150,7 +150,6 @@ public void startClient() {
         } catch (JMSException | IllegalArgumentException jmse) {
             LOG.error("Error while sending a message!", jmse.getMessage());
             jmse.printStackTrace();
-            System.exit(1);
```
Comment (jdanekrh) [inline jmslib/src/main/java/com/redhat/mqe/lib/SenderClient.java]: What is the goal there? This change does not have comment, and the commit message does not explain why this is being done either.

**Comment by jdanekrh** [comment]
Still the same situation as in my last comment. Some of this is in master, some is not. This branch contains essentially duplicate commits with the master branch.

@michalxo please rebase on top of master and choose to skip such commits, then force-push into the branch (force push is the only way to resolve this, as far as I know).

File: cli-artemis-jms/src/main/java/com/redhat/mqe/acc/AccSenderClient.java (hunk 2 above)
Comment (michalxo) [reply cli-artemis-jms/src/main/java/com/redhat/mqe/acc/AccSenderClient.java]: I think was with Core having some problem during failover (missing exactly 1 message).
Resend - for core protocol.

File: jmslib/src/main/java/com/redhat/mqe/lib/SenderClient.java (hunk 3 above)
Comment (michalxo) [reply jmslib/src/main/java/com/redhat/mqe/lib/SenderClient.java]: Most probably connected to the issue above - in failover scenarios, client should not get closed after JMSException (killed broker for example).

But I don't recall it properly TBH. :-/

**Comment by michalxo** [comment]
I will get back to this in few days/week(s).

**Comment by michalxo** [comment]
This was supposedly merged into master 12th of June.

**Pull request closed without merging**